src/env.cc

```diff
@@ -1045,6 +1048,9 @@ void Environment::RunAndClearInterrupts() {
 void Environment::RunAndClearNativeImmediates(bool only_refed) {
   TRACE_EVENT0(TRACING_CATEGORY_NODE1(environment),
                "RunAndClearNativeImmediates");
+  if (isolate_->IsExecutionTerminating())
```

Comment:
```suggestion
  if (is_stopping())
```

?

Reply:
I already tried that approach but it made a number of tests not to exit because, I think, they need that some immediates that don't call into JS to run before exiting (also, apparently calling `TerminateExecution()` doesn't make an immediate call to `IsExecutionTerminating()` to return true, it may take a while before it does).

Reply:
Well yeah, looking at this code, it seems like this isn’t the right place to just `return;` without calling the remaining callbacks. Some of them perform cleanup operations that shouldn’t be skipped, it’s rather up to the callbacks to ensure that they don’t call into JS.